lib/child_process.js

```diff
@@ -613,8 +596,37 @@ function normalizeSpawnArguments(file, args, options) {
 
 function spawn(file, args, options) {
   const child = new ChildProcess();
-
   options = normalizeSpawnArguments(file, args, options);
+
+  if (options.signal) {
+    const signal = options.signal;
+    // Validate signal, if present
+    validateAbortSignal(signal, 'options.signal');
+
+    // Do nothing and throw if already aborted
+    if (signal.aborted) {
+      onAbortListener();
+    } else {
+      const config = { once: true };
+      const remove = () => {
+        signal.removeEventListener('abort', onAbortListener, config);
+      };
+
+      signal.addEventListener('abort', onAbortListener, config);
+      child.once('close', remove);
```

Comment:
nit: we could also get rid of the `remove` variable.
```suggestion
      child.once('close',
                 () => signal.removeEventListener('abort', onAbortListener));
```

Reply:
Done in 72ff68a
Please have a look 🙂